libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -95,81 +93,73 @@ public void trackEventTest() {
 
-        Mockito.doAnswer(invocation -> {
-            RemoteDependencyTelemetry remoteDependencyTelemetry = invocation.getArgument(0);
+        Mockito.verify(mockTelemetryChannel, invocations -> {
+            RemoteDependencyTelemetry remoteDependencyTelemetry = invocations.getAllInvocations().get(0).getArgument(0);
 
             Assert.assertEquals("test", remoteDependencyTelemetry.getType());
             Assert.assertEquals("target", remoteDependencyTelemetry.getTarget());
             Assert.assertEquals("dependencyname", remoteDependencyTelemetry.getName());
             Assert.assertEquals("result", remoteDependencyTelemetry.getResultCode());
             Assert.assertFalse(remoteDependencyTelemetry.getSuccess());
-
-            return null;
-        }).when(mockTelemetryChannel).send(Mockito.any(RemoteDependencyTelemetry.class));
+        }).send(Mockito.any(AvailabilityTelemetry.class));
     }
 
     @Test
     public void trackExceptionTest() {
-        Exception expectedException = new Exception("test-exception");        
+        Exception expectedException = new Exception("test-exception");
         Map<String, String> properties = new HashMap<>();
         properties.put("foo", "bar");
         Map<String, Double> metrics = new HashMap<>();
         metrics.put("metric", 0.6);
-                
+
         botTelemetryClient.trackException(expectedException, properties, metrics);
 
-        Mockito.doAnswer(invocation -> {
-            ExceptionTelemetry exceptionTelemetry = invocation.getArgument(0);
+        Mockito.verify(mockTelemetryChannel, invocations -> {
+            ExceptionTelemetry exceptionTelemetry = invocations.getAllInvocations().get(0).getArgument(0);
 
             Assert.assertEquals(expectedException, exceptionTelemetry.getException());
             Assert.assertEquals("bar", exceptionTelemetry.getProperties().get("foo"));
             Assert.assertEquals(0, Double.compare(0.6, exceptionTelemetry.getMetrics().get("metric")));
-
-            return null;
-        }).when(mockTelemetryChannel).send(Mockito.any(ExceptionTelemetry.class));
+        }).send(Mockito.any(ExceptionTelemetry.class));
     }
 
     @Test
     public void trackTraceTest() {
         Map<String, String> properties = new HashMap<>();
         properties.put("foo", "bar");
-        
+
         botTelemetryClient.trackTrace("hello", Severity.CRITICAL, properties);
 
-        Mockito.doAnswer(invocation -> {
-            TraceTelemetry traceTelemetry = invocation.getArgument(0);
+        Mockito.verify(mockTelemetryChannel, invocations -> {
+            TraceTelemetry traceTelemetry = invocations.getAllInvocations().get(0).getArgument(0);
 
             Assert.assertEquals("hello", traceTelemetry.getMessage());
-            Assert.assertEquals(Severity.CRITICAL, traceTelemetry.getSeverityLevel());
+            Assert.assertEquals(Severity.CRITICAL.name(), traceTelemetry.getSeverityLevel().name().toUpperCase());
```

Comment:
Mm not quite sure about this. We have 2 different types of Severity, one local and another of the ApplicationInsights-Java library.

Check the internal migration (I guess that `BotTelemetryClientImpl`) that contains the mapping between both `enums`

Reply:
I don't know what you mean with this. Should I use a Severity `enum` from our migration (local)?

Reply:
Check C#, and also check the `BotTelemetryClientImpl`, if you have any doubt after, lets pair